L1Trigger/L1TMuonEndCap/src/PrimitiveSelection.cc

```diff
@@ -520,38 +483,26 @@ int PrimitiveSelection::select_csc(const TriggerPrimitive& muon_primitive) const
     int tp_bx        = tp_data.bx;
     int tp_csc_ID    = tp_data.cscID;
 
-    if ( !(emtf::MIN_ENDCAP <= tp_endcap && tp_endcap <= emtf::MAX_ENDCAP) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_endcap = "  << tp_endcap; return selected; }
-    if ( !(emtf::MIN_TRIGSECTOR <= tp_sector && tp_sector <= emtf::MAX_TRIGSECTOR) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_sector = "  << tp_sector; return selected; }
-    if ( !(1 <= tp_station && tp_station <= 4) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_station = " << tp_station; return selected; }
-    if ( !(1 <= tp_csc_ID && tp_csc_ID <= 9) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_csc_ID = "  << tp_csc_ID; return selected; }
-    if ( !(tp_data.strip < 160) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_data.strip = "   << tp_data.strip ; return selected; }
-    if ( !(tp_data.keywire < 128) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_data.keywire = " << tp_data.keywire; return selected; }
-    if ( !(tp_data.valid == true) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_data.valid = "   << tp_data.valid ; return selected; }
-    if ( !(tp_data.pattern <= 10) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_data.pattern = " << tp_data.pattern; return selected; }
-    if ( !(tp_data.quality > 0) ) {
-      edm::LogWarning("L1T") << "EMTF CSC format error: tp_data.quality = " << tp_data.quality; return selected; }
+    assert_no_abort(emtf::MIN_ENDCAP <= tp_endcap && tp_endcap <= emtf::MAX_ENDCAP);
```

Comment:
Could you define an "assert_and_abort" and use that for all the "assert" statements above, if you prefer not to have "LogError" all over the code?

Reply:
'assert_and_abort' would just be 'assert' though.

Reply:
As we discussed on Skype, "assert_and_abort" could call edm::LogError("L1T"), so that all run-time aborts are via edm.  @rekovic would know better than I do whether this is important.

Reply:
Sorry I might have missed that in my earlier reply. Did you mean you want all of the above (old lines 523-540) changed to LogError, instead of LogWarning? I thought you didn't want them to abort?

Reply:
Sorry, the comment is a bit confusing.  I was not suggesting that you change any of these lines.  This was the first place I had seen your "assert_no_abort" function, so after looking up the definition, I had the idea to create as well an "assert_and_abort" function using edm::LogError (or cms::Exception) which would replace the simple "assert" statements used everywhere else in the code.

Reply:
Ok, I will create the "assert_and_log_error" function. I think I wouldn't call it "assert_and_abort" because that's already what "assert" does. I would still prefer to use "assert" for conditions that should not fail even once (what I think about as coding/logic errors). It is a standard practice, and it is widely used in the C++ standard library (e.g. std::vector::at()). But I agree that some of them could be changed to use the new "assert_and_log_error".